presto-hive/src/main/java/com/facebook/presto/hive/HivePageSink.java

```diff
@@ -283,7 +284,7 @@ private HiveRecordWriter createWriter(List<Object> partitionRow)
         checkArgument(partitionRow.size() == partitionColumnNames.size(), "size of partitionRow is different from partitionColumnNames");
 
         List<String> partitionValues = partitionRow.stream()
-                .map(Object::toString) // todo this seems wrong
+                .map(Objects::toString)
```

Comment:
We should verify Hive's behavior. Create partitioned tables in Hive with partition columns of all the supported types (easiest is a separate table per type), then run insert queries with dynamic partitions that produce null and see what value Hive chooses for the partition key. Make sure the null is read from another table and is not a constant, as Hive sometimes has different behavior for constants.